fiftyone/utils/sam.py

```diff
@@ -233,6 +233,20 @@ def _forward_pass_boxes(self, imgs):
 
         outputs = []
         for img, detections in zip(imgs, self._curr_prompts):
+            ## If no detections, return empty tensors instead of running SAM
+            if (
+                detections.detections is None
```

Comment:
Ah, the proper check is:

```py
detections is None or len(detections.detections) == 0
```

`detections.detections` will never be `None`, but `detections` itself could be `None` if the sample hasn't had the field populated.